ogr/ogrsf_frmts/gpkg/ogrgeopackagedriver.cpp

```diff
@@ -246,6 +246,88 @@ static int OGRGeoPackageDriverIdentify(GDALOpenInfo *poOpenInfo)
     return OGRGeoPackageDriverIdentify(poOpenInfo, osIgnored, false);
 }
 
+/************************************************************************/
+/*                    OGRGeoPackageDriverGetSubdatasetInfo()            */
+/************************************************************************/
+
+struct OGRGeoPackageDriverSubdatasetInfo : public GDALSubdatasetInfo
+{
+    virtual ~OGRGeoPackageDriverSubdatasetInfo() = default;
+
+    bool IsSubdatasetSyntax(const std::string &fileName) const override
```

Comment:
seeing this implementation indeed makes me thinks that my above remark about pfnGetSubdatasetInfoFunc taking the subdataset name to be probably better, as saving lines of code. The components of the subdataset could be saved as member variable(s) of the class during construction

The advantage of your approach is that it enables to deal with several subdatasets with a same GDALSubdatasetInfo instance, but I don't think that's a real use case. And the name SubdatasetInfo would make me think that this is a class whose instances know about a particular subdataset

Reply:
Sounds good, I had the same doubt but reading the issue comments it wasn't totally clear which way you wanted to go.